### PR TITLE
Created TikTok meric scraper.

### PR DESCRIPTION
**Pull request by bernardbaker**
Only testing for success and failure in the network data response.

Test instructions are in the README.md.

### CODE REVIEW

**Review by un7c0rn — CHANGES_REQUESTED** [review]
Overall looks great. A few notes:

1. The instructions reference port 8888 but it seems that by default port 3000 is used. I worked around this by adding "start": "PORT=8888 run-p start:*" to package.json but not sure that's the right thing to do here. Please advise.

2. I'm a little confused about the role netlify is playing. I see a lambdas folder that seems to imply the metric parsing logic would be running server side? Ideally we perform metric parsing locally so that our API calls to Tik Tok are distributed across client devices.

3. When I follow the instructions I'm seeing attached. Is this expected?

https://snipboard.io/ZItJqH.jpg

**Comment by bernardbaker** [comment]
> The instructions reference port 8888 but it seems that by default port 3000 is used. I worked around this by adding "start": "PORT=8888 run-p start:*" to package.json but not sure that's the right thing to do here. Please advise.

The README.md needs to be corrected. I'll update it.

> I'm a little confused about the role netlify is playing. I see a lambdas folder that seems to imply the metric parsing logic would be running server side? Ideally we perform metric parsing locally so that our API calls to Tik Tok are distributed across client devices.

Netlify provides a serverless API layer to the thin client. The TikTok library mentioned that there isn't a limit on the number of requests that can be made.

> When I follow the instructions I'm seeing attached. Is this expected?
Looks like the `./src/functions` files haven't been built. You'll also need to rerun the tests.

**Review by un7c0rn — APPROVED** [review]
Looks good!

**Comment by un7c0rn** [comment]
rebased and reopened to get PR builder to run.

**Comment by bernardbaker** [comment]
Keeping commits messages.

File: react_front_end/src/__tests__/services/TikTok.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+/**
+ * @jest-environment node
+ */
+
+import { metricParser } from "../../services/TikTok";
+
+test("video metadata", async () => {
+  const url = "https://www.tiktok.com/@tiktok/video/6807491984882765062";
+  const data = await metricParser(url);
+  expect(data.id).toBe("6807491984882765062");
```
Comment (dzx11) [inline react_front_end/src/__tests__/services/TikTok.js:10]: @bernardbaker this test is failing for me right now when I run 'yarn test', "cannot read property 'id' of undefined", see below. Is this supposed to happen? 
test

Comment (bernardbaker) [reply react_front_end/src/__tests__/services/TikTok.js:10]: @dzx11 Could you provide the full stack trace. From the console.

Comment (bernardbaker) [reply react_front_end/src/__tests__/services/TikTok.js:10]: @dzx11 Also try increasing the `jest.setTimeout(n)`.

https://jestjs.io/docs/en/jest-object#jestsettimeouttimeout

Comment (dzx11) [reply react_front_end/src/__tests__/services/TikTok.js:10]: @bernardbaker It looks like it's trying to connect to localhost:8888 for some reason. Are there some config changes that need to be made to change this? I think I saw some discussion with Brandon regarding this. Using jest.setTimeout didn't make a difference.
Full trace below.
test2

**Comment by bernardbaker** [comment]
Did you follow the directions in the README.md on how to start the project?

Regards,

Bernard T. A. Baker

**Comment by dzx11** [comment]
@bernardbaker I reran the install scripts, changed the port in the test script, and tried running the 'yarn start' script concurrently and still had an error, although the console error messages are different. How are you running the tests? Do you just run the 'yarn test' script or do you have another script or server running at the same time? Since it's trying to connect to localhost, I assume that you're running the 'yarn start' script at the same time as you run the tests
test4

**Comment by bernardbaker** [comment]
@dzx11 try this...

open three terminal windows.

1st window: cd react_front_end; yarn
2nd window after 1st set of commands have completed: netlify dev
3rd window after 2nd set of commands have completed: yarn test

Regards,

Bernard T. A. Baker

**Comment by dzx11** [comment]
@bernardbaker that worked, thanks! It may be worth adding/clarifying those steps to the readme - I was under the impression that netlify was only needed for deployment.

**Comment by bernardbaker** [comment]
@dzx11 great news. Good job. I'll update the README.md as requested.

**Pull request merged**